modules/elasticsearch-index/src/main/java/org/opencastproject/elasticsearch/index/endpoint/IndexEndpoint.java

```diff
@@ -152,6 +153,57 @@ public Response partiallyRebuildIndex(@PathParam("service") final String service
     return R.ok();
   }
 
+  @POST
+  @Path("rebuild/AssetManager/{part}")
+  @RestQuery(name = "partiallyRebuildIndexByPart",
```

Comment:
While I generally agree, since this is not the main focus of this PR and I'm limited on time, I tried to not put too much effort into this while hopefully keeping it easily extendable for the future. But if you say your suggestion should really be part of this PR (for example to avoid changing the endpoint path down the line), I'll get to it.

Reply:
I think it would be good to get this right from the get-go. Even though we don't guarantee anything for these endpoints, changing them sometime down the line is always a hassle.

Reply:
Tried to rewrite the endpoint. Currently confused by https://github.com/opencast/opencast/blob/ccfe8dd415bbeea3b1f8a8ab1bda45b4ff2b54d2/modules/elasticsearch-index/src/main/java/org/opencastproject/elasticsearch/index/rebuild/IndexRebuildService.java#L212, which eats up exceptions during index rebuilds for reasons unknown to me. Will try to look further into this some other time. Furthermore, checking ServicePart in IndexEndpoint is not really possible anymore if ServicePart is moved to the actual service due to code architecture, so currently the endpoint always returns 200, no matter if the part was valid or not.

Reply:
I can certainly throw an exception there, but this will not solve the problem that the index rebuild is running async (I think?), so I can't have it influence the return code.

Reply:
Ah, of course, you're right about that. Sorry, that slipped my mind. Well, at least the UI will show you if it went wrong... IMO that exception should still be thrown, though. Adding to the list... I should stop looking at that code.